src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MemoryHelpers.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// These methods are used to implement memcpy and memset intrinsics with null checks.
+    /// </summary>
+    internal static class MemoryHelpers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void MemSet(byte* dest, int value, nuint size)
+        {
+            if (size > 0)
+            {
+                _ = *dest;
+                RuntimeImports.memset(dest, value, size);
```

Comment:
This can call `SpanHelpers.Fill` / `Buffer.Memmove` instead of the C runtime APIs. The managed equivalents are more friendly to GC suspension.

Reply:
Done